llvm/lib/Transforms/IPO/FunctionImport.cpp

```diff
@@ -174,6 +174,17 @@ static cl::opt<std::string> WorkloadDefinitions(
              "}"),
     cl::Hidden);
 
+static cl::opt<bool> ImportAssumeUniqueLocal(
+    "import-assume-unique-local", cl::init(false),
+    cl::desc(
+        "By default, a local-linkage global variable won't be imported in the "
+        "edge mod1:func -> mod2:local-var (from value profiles) since compiler "
+        "cannot assume mod2 is compiled with full path which gives local-var a "
+        "program-wide unique GUID. Set this option to true will help cross "
+        "module import of such variables. This is only safe if the compiler "
```

Comment:
I am trying to learn more about this space. Can you elaborate on "This is only safe if the compiler user specify the full module path." How does user specify it? I looked at the test and it's not clear to me from it.

Reply:
> How does user specify it?

In a monolithic repository (https://research.google/pubs/why-google-stores-billions-of-lines-of-code-in-a-single-repository/) of source code, user specifies the full path in the compile options.

Using the folder structure below as an example [1], `clang -O2 <other-options> /tmp/src1/a.cpp /tmp/src1/b.cpp /tmp/src2/a.cpp /tmp/src2/b.cpp /tmp/src2/c.cpp` gives the full path, and `cd /tmp/src1; clang -O2 a.cpp b.cpp <other arguments to specify paths in src2>` doesn't.

In practice, bazel (https://bazel.build/) configures the build rules and guarantees full path is used. Users use bazel to compile an executable.

>  I looked at the test and it's not clear to me from it.

The purpose of the test is to illustrate and provide coverage for profile-gen and its use, so I saved the extra work to use full path there.

[1]
```
/tmp/
|── src1/
│   ├── a.cpp
│   ├── b.cpp
├── src2/
│   ├── a.cpp
│   ├── b.cpp
│   └──c.cpp
```

Reply:
The test program itself is correct as there are only two files and they don't have conflict names.

I'm open to update it to use the full path if that is preferred or better.

Reply:
FYI, I double checked whether compiler correctness can rely on the fact that full file paths are used. My takeaway is that bazel doesn't fail the executable compilation if local paths are used (if any in a unconventional way) so I don't plan to turn on this option.

Yet still `NumDefs == 1` condition (https://github.com/llvm/llvm-project/pull/100448/files#diff-e7cb370fee46f0f773f2b5429dfab36b75126d3909ae98ee87ff3d0e3f75c6e9R215) should hold true for many internal-linkage vtables as long as full paths are indeed used.

Reply:
Gotcha, thx for explanation.